google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -179,6 +206,99 @@ public Builder setSessionLabels(Map<String, String> sessionLabels) {
       return this;
     }
 
+    /**
+     * {@link SpannerOptions.Builder} does not support global retry settings, as it creates three
+     * different gRPC clients: {@link Spanner}, {@link DatabaseAdminClient} and {@link
+     * InstanceAdminClient}. Instead of calling this method, you should set specific {@link
+     * RetrySettings} for each of the underlying gRPC clients by calling respectively {@link
+     * #spannerStubSettingsBuilder()}, {@link #databaseAdminStubSettingsBuilder()} or {@link
+     * #instanceAdminStubSettingsBuilder()}.
+     */
+    @Override
+    public Builder setRetrySettings(RetrySettings retrySettings) {
+      throw new UnsupportedOperationException(
+          "SpannerOptions does not support setting global retry settings. "
+              + "Call spannerStubSettingsBuilder().<method-name>Settings().setRetrySettings(RetrySettings) instead.");
+    }
+
+    /**
+     * Returns the {@link SpannerStubSettings.Builder} that will be used to build the {@link
+     * SpannerRpc}. Use this to set custom {@link RetrySettings} for individual gRPC methods.
+     *
+     * <p>The library will automatically use sensible defaults if no custom settings are set. The
+     * defaults are the same as the defaults that are used by {@link SpannerSettings}. Retries are
+     * configured for idempotent methods but not for non-idempotent methods.
+     *
+     * <p>You can set the same {@link RetrySettings} for all unary methods by calling this:
+     *
+     * <pre><code>
+     * builder
+     *     .spannerStubSettingsBuilder()
+     *     .applyToAllUnaryMethods(
+     *         new ApiFunction&lt;UnaryCallSettings.Builder&lt;?, ?&gt;, Void&gt;() {
+     *           public Void apply(Builder&lt;?, ?&gt; input) {
+     *             input.setRetrySettings(retrySettings);
+     *             return null;
+     *           }
+     *         });
+     * </code></pre>
+     */
+    public SpannerStubSettings.Builder spannerStubSettingsBuilder() {
```

Comment:
I think I would prefix 'new' to all of the builder names to be consistent with other areas in the library.

Reply:
ex. `newSpannerStubSettingsBuilder()`

Reply:
sorry - looking at this again and realize it's returning a variable. Should we just return a new builder here instead, or is there a reason why we are storing it?